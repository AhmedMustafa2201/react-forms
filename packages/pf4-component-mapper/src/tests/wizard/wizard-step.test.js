import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import WizardStep from '../../form-fields/wizard/wizard-step';
import FieldProvider from '../../../../../__mocks__/mock-field-provider';

describe('<WizardStep />', () => {
  let initialProps;
  beforeEach(() => {
    initialProps = {
      fields: [ 'foo' ],
      formOptions: {
        renderForm: item => <div key={ item }>{ item }</div>,
        onCancel: jest.fn(),
        handleSubmit: jest.fn(),
      },
      FieldProvider,
      handlePrev: jest.fn(),
      handleNext: jest.fn(),
      buttonLabels: {
        cancel: 'Cancel',
        next: 'Next',
        submit: 'Submit',
        back: 'Back',
      },
      title: 'title',
      description: 'description',
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<WizardStep  { ...initialProps }/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render custom title', () => {
    const wrapper = shallow(<WizardStep  { ...initialProps } title={ <div>title</div> }/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render custom description', () => {
    const wrapper = shallow(<WizardStep  { ...initialProps } description={ <div>description</div> }/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
