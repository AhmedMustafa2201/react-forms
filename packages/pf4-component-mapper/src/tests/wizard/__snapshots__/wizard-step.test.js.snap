// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WizardStep /> should render correctly 1`] = `
<ReactFinalForm
  onSubmit={[MockFunction]}
>
  <WizardStep
    StepTemplate={[Function]}
    buttonLabels={
      Object {
        "back": "Back",
        "cancel": "Cancel",
        "next": "Next",
        "submit": "Submit",
      }
    }
    description="description"
    fields={
      Array [
        "foo",
      ]
    }
    formOptions={
      Object {
        "getState": [MockFunction],
        "handleSubmit": [MockFunction],
        "onCancel": [MockFunction],
        "renderForm": [Function],
      }
    }
    handleNext={[MockFunction]}
    handlePrev={[MockFunction]}
    title="title"
  >
    <Component>
      <main
        className="pf-c-wizard__main"
      >
        <div
          className="pf-c-wizard__main-body"
        >
          <DefaultStepTemplate
            buttonLabels={
              Object {
                "back": "Back",
                "cancel": "Cancel",
                "next": "Next",
                "submit": "Submit",
              }
            }
            description="description"
            fields={
              Array [
                "foo",
              ]
            }
            formFields={
              Array [
                <div>
                  foo
                </div>,
              ]
            }
            formOptions={
              Object {
                "getState": [MockFunction],
                "handleSubmit": [MockFunction],
                "onCancel": [MockFunction],
                "renderForm": [Function],
              }
            }
            formRef={
              Object {
                "current": <div
                  class="pf-c-form"
                >
                  <div>
                    foo
                  </div>
                </div>,
              }
            }
            handleNext={[MockFunction]}
            handlePrev={[MockFunction]}
            title="title"
          >
            <div
              className="pf-c-form"
            >
              <div
                key="foo"
              >
                foo
              </div>
            </div>
          </DefaultStepTemplate>
        </div>
      </main>
    </Component>
    <WizardStepButtons
      buttonLabels={
        Object {
          "back": "Back",
          "cancel": "Cancel",
          "next": "Next",
          "submit": "Submit",
        }
      }
      formOptions={
        Object {
          "getState": [MockFunction],
          "handleSubmit": [MockFunction],
          "onCancel": [MockFunction],
          "renderForm": [Function],
        }
      }
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
    >
      <footer
        className="pf-c-wizard__footer "
      >
        <FormSpy>
          <NextButton
            getState={[MockFunction]}
            handleNext={[MockFunction]}
            handleSubmit={[MockFunction]}
            nextLabel="Next"
            onCancel={[MockFunction]}
            renderForm={[Function]}
            submitLabel="Submit"
          >
            <Component
              isDisabled={true}
              onClick={[Function]}
              type="button"
              variant="primary"
            >
              <button
                aria-disabled={null}
                aria-label={null}
                className="pf-c-button pf-m-primary"
                data-ouia-component-id={null}
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={true}
                onClick={[Function]}
                tabIndex={null}
                type="button"
              >
                Submit
              </button>
            </Component>
          </NextButton>
          <Component
            onClick={[MockFunction]}
            type="button"
            variant="secondary"
          >
            <button
              aria-disabled={null}
              aria-label={null}
              className="pf-c-button pf-m-secondary"
              data-ouia-component-id={null}
              data-ouia-component-type="PF4/Button"
              data-ouia-safe={true}
              disabled={false}
              onClick={[MockFunction]}
              tabIndex={null}
              type="button"
            >
              Back
            </button>
          </Component>
          <Component
            onClick={[MockFunction]}
            type="button"
            variant="link"
          >
            <button
              aria-disabled={null}
              aria-label={null}
              className="pf-c-button pf-m-link"
              data-ouia-component-id={null}
              data-ouia-component-type="PF4/Button"
              data-ouia-safe={true}
              disabled={false}
              onClick={[MockFunction]}
              tabIndex={null}
              type="button"
            >
              Cancel
            </button>
          </Component>
        </FormSpy>
      </footer>
    </WizardStepButtons>
  </WizardStep>
</ReactFinalForm>
`;

exports[`<WizardStep /> should render custom description 1`] = `
<ReactFinalForm
  onSubmit={[MockFunction]}
>
  <WizardStep
    StepTemplate={[Function]}
    buttonLabels={
      Object {
        "back": "Back",
        "cancel": "Cancel",
        "next": "Next",
        "submit": "Submit",
      }
    }
    description={
      <div>
        description
      </div>
    }
    fields={
      Array [
        "foo",
      ]
    }
    formOptions={
      Object {
        "getState": [MockFunction],
        "handleSubmit": [MockFunction],
        "onCancel": [MockFunction],
        "renderForm": [Function],
      }
    }
    handleNext={[MockFunction]}
    handlePrev={[MockFunction]}
    title="title"
  >
    <Component>
      <main
        className="pf-c-wizard__main"
      >
        <div
          className="pf-c-wizard__main-body"
        >
          <DefaultStepTemplate
            buttonLabels={
              Object {
                "back": "Back",
                "cancel": "Cancel",
                "next": "Next",
                "submit": "Submit",
              }
            }
            description={
              <div>
                description
              </div>
            }
            fields={
              Array [
                "foo",
              ]
            }
            formFields={
              Array [
                <div>
                  foo
                </div>,
              ]
            }
            formOptions={
              Object {
                "getState": [MockFunction],
                "handleSubmit": [MockFunction],
                "onCancel": [MockFunction],
                "renderForm": [Function],
              }
            }
            formRef={
              Object {
                "current": <div
                  class="pf-c-form"
                >
                  <div>
                    foo
                  </div>
                </div>,
              }
            }
            handleNext={[MockFunction]}
            handlePrev={[MockFunction]}
            title="title"
          >
            <div
              className="pf-c-form"
            >
              <div
                key="foo"
              >
                foo
              </div>
            </div>
          </DefaultStepTemplate>
        </div>
      </main>
    </Component>
    <WizardStepButtons
      buttonLabels={
        Object {
          "back": "Back",
          "cancel": "Cancel",
          "next": "Next",
          "submit": "Submit",
        }
      }
      formOptions={
        Object {
          "getState": [MockFunction],
          "handleSubmit": [MockFunction],
          "onCancel": [MockFunction],
          "renderForm": [Function],
        }
      }
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
    >
      <footer
        className="pf-c-wizard__footer "
      >
        <FormSpy>
          <NextButton
            getState={[MockFunction]}
            handleNext={[MockFunction]}
            handleSubmit={[MockFunction]}
            nextLabel="Next"
            onCancel={[MockFunction]}
            renderForm={[Function]}
            submitLabel="Submit"
          >
            <Component
              isDisabled={true}
              onClick={[Function]}
              type="button"
              variant="primary"
            >
              <button
                aria-disabled={null}
                aria-label={null}
                className="pf-c-button pf-m-primary"
                data-ouia-component-id={null}
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={true}
                onClick={[Function]}
                tabIndex={null}
                type="button"
              >
                Submit
              </button>
            </Component>
          </NextButton>
          <Component
            onClick={[MockFunction]}
            type="button"
            variant="secondary"
          >
            <button
              aria-disabled={null}
              aria-label={null}
              className="pf-c-button pf-m-secondary"
              data-ouia-component-id={null}
              data-ouia-component-type="PF4/Button"
              data-ouia-safe={true}
              disabled={false}
              onClick={[MockFunction]}
              tabIndex={null}
              type="button"
            >
              Back
            </button>
          </Component>
          <Component
            onClick={[MockFunction]}
            type="button"
            variant="link"
          >
            <button
              aria-disabled={null}
              aria-label={null}
              className="pf-c-button pf-m-link"
              data-ouia-component-id={null}
              data-ouia-component-type="PF4/Button"
              data-ouia-safe={true}
              disabled={false}
              onClick={[MockFunction]}
              tabIndex={null}
              type="button"
            >
              Cancel
            </button>
          </Component>
        </FormSpy>
      </footer>
    </WizardStepButtons>
  </WizardStep>
</ReactFinalForm>
`;

exports[`<WizardStep /> should render title with showTitle 1`] = `
<ReactFinalForm
  onSubmit={[MockFunction]}
>
  <WizardStep
    StepTemplate={[Function]}
    buttonLabels={
      Object {
        "back": "Back",
        "cancel": "Cancel",
        "next": "Next",
        "submit": "Submit",
      }
    }
    description="description"
    fields={
      Array [
        "foo",
      ]
    }
    formOptions={
      Object {
        "getState": [MockFunction],
        "handleSubmit": [MockFunction],
        "onCancel": [MockFunction],
        "renderForm": [Function],
      }
    }
    handleNext={[MockFunction]}
    handlePrev={[MockFunction]}
    showTitle={true}
    title={
      <div>
        title
      </div>
    }
  >
    <Component>
      <main
        className="pf-c-wizard__main"
      >
        <div
          className="pf-c-wizard__main-body"
        >
          <DefaultStepTemplate
            buttonLabels={
              Object {
                "back": "Back",
                "cancel": "Cancel",
                "next": "Next",
                "submit": "Submit",
              }
            }
            description="description"
            fields={
              Array [
                "foo",
              ]
            }
            formFields={
              Array [
                <div>
                  foo
                </div>,
              ]
            }
            formOptions={
              Object {
                "getState": [MockFunction],
                "handleSubmit": [MockFunction],
                "onCancel": [MockFunction],
                "renderForm": [Function],
              }
            }
            formRef={
              Object {
                "current": <div
                  class="pf-c-form"
                >
                  <h1
                    class="pf-c-title pf-m-xl"
                  >
                    <div>
                      title
                    </div>
                  </h1>
                  <div>
                    foo
                  </div>
                </div>,
              }
            }
            handleNext={[MockFunction]}
            handlePrev={[MockFunction]}
            showTitle={true}
            title={
              <div>
                title
              </div>
            }
          >
            <div
              className="pf-c-form"
            >
              <RenderTitle
                title={
                  <div>
                    title
                  </div>
                }
              >
                <Component
                  headingLevel="h1"
                  size="xl"
                >
                  <h1
                    className="pf-c-title pf-m-xl"
                  >
                    <div>
                      title
                    </div>
                  </h1>
                </Component>
              </RenderTitle>
              <div
                key="foo"
              >
                foo
              </div>
            </div>
          </DefaultStepTemplate>
        </div>
      </main>
    </Component>
    <WizardStepButtons
      buttonLabels={
        Object {
          "back": "Back",
          "cancel": "Cancel",
          "next": "Next",
          "submit": "Submit",
        }
      }
      formOptions={
        Object {
          "getState": [MockFunction],
          "handleSubmit": [MockFunction],
          "onCancel": [MockFunction],
          "renderForm": [Function],
        }
      }
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
    >
      <footer
        className="pf-c-wizard__footer "
      >
        <FormSpy>
          <NextButton
            getState={[MockFunction]}
            handleNext={[MockFunction]}
            handleSubmit={[MockFunction]}
            nextLabel="Next"
            onCancel={[MockFunction]}
            renderForm={[Function]}
            submitLabel="Submit"
          >
            <Component
              isDisabled={true}
              onClick={[Function]}
              type="button"
              variant="primary"
            >
              <button
                aria-disabled={null}
                aria-label={null}
                className="pf-c-button pf-m-primary"
                data-ouia-component-id={null}
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={true}
                onClick={[Function]}
                tabIndex={null}
                type="button"
              >
                Submit
              </button>
            </Component>
          </NextButton>
          <Component
            onClick={[MockFunction]}
            type="button"
            variant="secondary"
          >
            <button
              aria-disabled={null}
              aria-label={null}
              className="pf-c-button pf-m-secondary"
              data-ouia-component-id={null}
              data-ouia-component-type="PF4/Button"
              data-ouia-safe={true}
              disabled={false}
              onClick={[MockFunction]}
              tabIndex={null}
              type="button"
            >
              Back
            </button>
          </Component>
          <Component
            onClick={[MockFunction]}
            type="button"
            variant="link"
          >
            <button
              aria-disabled={null}
              aria-label={null}
              className="pf-c-button pf-m-link"
              data-ouia-component-id={null}
              data-ouia-component-type="PF4/Button"
              data-ouia-safe={true}
              disabled={false}
              onClick={[MockFunction]}
              tabIndex={null}
              type="button"
            >
              Cancel
            </button>
          </Component>
        </FormSpy>
      </footer>
    </WizardStepButtons>
  </WizardStep>
</ReactFinalForm>
`;
